xmlschema-core/src/main/java/org/apache/ws/commons/schema/XmlSchemaSerializer.java

```diff
@@ -72,6 +74,11 @@ public XmlSchemaSerializerException(String msg) {
     public static final String XSD_NAMESPACE = XMLConstants.W3C_XML_SCHEMA_NS_URI;
     private static final String XMLNS_NAMESPACE_URI = "http://www.w3.org/2000/xmlns/";
 
+    /**
+     * Keeps track of already serialized schemas.
+     */
+    private static ThreadLocal<Set<XmlSchema>> serializedSchemaElements = new ThreadLocal<>();
```

Comment:
Do we really need a `ThreadLocal` for this?

Reply:
I believe so.

When imports are used, new instances of `XmlSchemaSerializer` are created for them so we need to share the same `Set<XmlSchema>` between multiple instances of serializer. That's why `static` is used.

But we don't want to share the same `Set<XmlSchema>` globally between all threads that are potentially serializing something in the same JVM. That leads to `ThreadLocal`.